src/audio/mux/mux.c

```diff
@@ -234,53 +193,60 @@ static int build_config(struct comp_data *cd, struct comp_dev *dev)
 	}
 	return 0;
 }
+#endif
 
-static struct comp_dev *mux_new(const struct comp_driver *drv,
-				const struct comp_ipc_config *config,
-				const void *spec)
+static int mux_init(struct processing_module *mod)
 {
-	const struct mux_data *ipc_process = spec;
-	struct comp_dev *dev;
+	struct module_data *module_data = &mod->priv;
+	struct comp_dev *dev = mod->dev;
+	struct module_config *cfg = &module_data->cfg;
 	struct comp_data *cd;
 	int ret;
 
-	comp_cl_info(&comp_mux, "mux_new()");
+	comp_info(dev, "mux_init()");
 
-	dev = comp_alloc(drv, sizeof(*dev));
-	if (!dev)
-		return NULL;
-	dev->ipc_config = *config;
+	cd = rzalloc(SOF_MEM_ZONE_RUNTIME, 0, SOF_MEM_CAPS_RAM,
```

Comment:
Seems to be wrong size for cd, since config is a pointer with model handler for binary blob.

Reply:
Fixed in latest push, actually needed to revert back to struct instead of pointer.